nozzle/api_client.go

```diff
@@ -0,0 +1,103 @@
+package nozzle
+
+import (
+	"log"
+	"time"
+
+	cfclient "github.com/cloudfoundry-community/go-cfclient"
+	cache "github.com/patrickmn/go-cache"
+)
+
+// APIClient wrapper for Cloud Foundry Client
+type APIClient struct {
+	clientConfig *cfclient.Config
+	client       *cfclient.Client
+	appCache     *cache.Cache
+	logger       *log.Logger
+}
+
+// AppInfo holds Cloud Foundry applications information
+type AppInfo struct {
+	Name  string
+	Space string
+	Org   string
+}
+
+func newAppInfo(app cfclient.App) *AppInfo {
+	space, _ := app.Space()
+	org, _ := space.Org()
+	return &AppInfo{Name: app.Name, Space: space.Name, Org: org.Name}
+}
+
+// NewAPIClient crate a new ApiClient
+func NewAPIClient(conf *NozzleConfig, logger *log.Logger) (*APIClient, error) {
+	config := &cfclient.Config{
+		ApiAddress:        conf.APIURL,
+		Username:          conf.Username,
+		Password:          conf.Password,
+		SkipSslValidation: conf.SkipSSL,
+	}
+
+	client, err := cfclient.NewClient(config)
+	if err != nil {
+		return nil, err
+	}
+
+	return &APIClient{
+		clientConfig: config,
+		client:       client,
+		appCache:     cache.New(6*time.Hour, time.Hour),
+		logger:       logger,
+	}, nil
+}
+
+// FetchTrafficControllerURL return Doppler Endpoint URL
+func (api *APIClient) FetchTrafficControllerURL() string {
+	return api.client.Endpoint.DopplerEndpoint
+}
+
+// FetchAuthToken wrapper for client.GetToken()
+func (api *APIClient) FetchAuthToken() (string, error) {
```

Comment:
I didn't trace this all the way down, but, I'm not 100% sure on this implementation of fetching token. I just want to make sure that the refresh token is being properly handled. Nozzles are long lived processes and I've seen implementations that lose their auth because the refresh/expiration flow isn't handles correctly.

Not sure this is an issue, just flagging this as something to dig into to make sure that the token is correctly refreshed.

Reply:
Is there any way to force the token to expire?